src/Nuget.PackageIndex/Abstractions/FileSystem.cs

```diff
@@ -31,5 +31,20 @@ public DateTime FileGetLastWriteTime(string fullPath)
         {
             return File.GetLastWriteTime(fullPath);
         }
+
+        public string FileReadAllText(string fullPath)
+        {
+            return File.ReadAllText(fullPath);
+        }
+
+        public void FileWriteAllText(string fullPath, string content)
+        {
+            File.WriteAllText(fullPath, content);
+        }
+
+        public void FileDelete(string fullPath)
+        {
+            File.Delete(fullPath);
```

Comment:
The abstractions are exposing all the file operations directly, but all the exception flow through directly. Which means whoever uses the abstractions cannot rely on any contract for error handling other than knowing the underlying implementation.